### PR TITLE
[Table] States colors & checkbox alignement fixes

### PR DESCRIPTION
**Pull request by jeremie-lucca**
## Description

- Hover : grey 25 > grey 50
- Selected line : product 50
- Selected line hovered in a clickable table : product 100
- Fix checkbox alignement
- Replace old checkbox by the new one

-----
Capture d’écran 2024-01-29 à 14 49 40

https://github.com/LuccaSA/lucca-front/assets/25581936/95ac6353-d90f-4d64-8ff1-9f0f814b046c

https://github.com/LuccaSA/lucca-front/assets/25581936/f899966d-1693-41eb-a723-6e0a24128bbd

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2540/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/12430/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2540/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/12441/ui-diff)

File: packages/scss/src/components/table/mods.scss [hunk 1]
```diff
@@ -216,6 +216,27 @@
 			width: var(--spacings-S);
 		}
 	}
+
+	&:has(input[type='checkbox']:checked) {
+		.table-head-row-cell,
+		.table-body-row-cell,
+		.table-foot-row-cell {
+			background: var(--palettes-primary-50);
```
Comment (vvalentin-lucca) [inline packages/scss/src/components/table/mods.scss:224]:
```suggestion
			background-color: var(--palettes-primary-50);
```

File: stories/qa/table/table.stories.html [hunk 2]
```diff
@@ -76,14 +76,14 @@ <h2>Parents</h2>
...
+	<table class="table mod-clickable">
 		<thead class="table-head">
 			<tr class="table-head-row mod-selectable">
 				<th class="table-head-row-cell">
-					<div class="checkbox mod-noLabel">
-						<input class="checkbox-input" type="checkbox" id="rowAll" />
-						<label class="checkbox-label" for="rowAll"></label>
-					</div>
+					<span class="checkboxField">
+						<input type="checkbox" class="checkboxField-input" id="CBALL" aria-describedby="CBALLmessage" aria-required="true" />
```
Comment (vvalentin-lucca) [inline stories/qa/table/table.stories.html:84]: label with u-mask?

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2540/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/12821/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2540/storybook

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2540/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/12827/ui-diff)

**Review by vvalentin-lucca — APPROVED** [review]

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2540/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/12845/ui-diff)

**Pull request merged**